addons/spreadsheet/static/src/global_filters/helpers.js

```diff
@@ -334,6 +352,213 @@ export function getRelativeDateFromTo(now, offset, period) {
+        case "range":
+            return getPreviousRangeDateFilterValue(value);
+    }
+    return undefined;
+}
+
+/**
+ * Compute the next relative date filter value.
+ *
+ * @param {RelativeDateValue} value
+ * @returns {RelativeDateValue}
+ */
+function getNextRelativeDateFilterValue(value) {
+    switch (value.period) {
+        case "today":
+        case "yesterday":
+        case "last_7_days":
+        case "last_30_days":
+        case "last_90_days": {
+            const { from, to } = getRelativeDateFromTo(DateTime.local(), 1, value.period);
+            return {
+                type: "range",
+                from: from.toISODate(),
+                to: to.toISODate(),
+            };
+        }
+        case "last_12_months": {
+            const { from, to } = getRelativeDateFromTo(DateTime.local(), 1, value.period);
+            return {
+                type: "range",
+                from: from.startOf("month").toISODate(),
+                to: to.endOf("month").toISODate(),
+            };
+        }
+        case "last_month": {
+            const now = DateTime.local();
+            return {
+                type: "month",
+                year: now.year,
+                month: now.month,
+            };
+        }
+        case "month_to_date": {
+            const now = DateTime.local().plus({ months: 1 });
+            return {
+                type: "month",
+                year: now.year,
+                month: now.month,
+            };
+        }
+        case "year_to_date": {
+            return {
+                type: "year",
+                year: DateTime.local().year + 1,
+            };
+        }
+    }
+}
+
+/**
+ * Compute the previous relative date filter value.
+ *
+ * @param {RelativeDateValue} value
+ * @returns {RelativeDateValue}
+ */
+function getPreviousRelativeDateFilterValue(value) {
+    switch (value.period) {
+        case "today":
+        case "yesterday":
+        case "last_7_days":
+        case "last_30_days":
+        case "last_90_days": {
+            const { from, to } = getRelativeDateFromTo(DateTime.local(), -1, value.period);
+            return {
+                type: "range",
+                from: from.toISODate(),
+                to: to.toISODate(),
+            };
+        }
+        case "last_12_months": {
+            const { from, to } = getRelativeDateFromTo(DateTime.local(), -1, value.period);
+            return {
+                type: "range",
+                from: from.startOf("month").toISODate(),
+                to: to.endOf("month").toISODate(),
+            };
+        }
+
+        case "last_month": {
+            const now = DateTime.local().minus({ months: 2 });
+            return {
+                type: "month",
+                year: now.year,
+                month: now.month,
+            };
+        }
+        case "month_to_date": {
+            const now = DateTime.local().minus({ months: 1 });
```

Comment:
would it be an idea to keep the same date but the previous month ?
If today we are the 19th June, `month_to_date` gives 1st June - 19 June
The previous would be 1st May - 19 May. Does it makes sense? Of course, there are weekends etc in the way but...
It would allow to better compare months and answer questions like: *"after 19 days last month I had X brols"*